lib/Doctrine/Common/Cache/PhpFileCache.php

```diff
@@ -78,23 +78,20 @@ protected function doSave($id, $data, $lifeTime = 0)
             $lifeTime = time() + $lifeTime;
         }
 
-        if (is_object($data) && ! method_exists($data, '__set_state')) {
-            throw new \InvalidArgumentException(
-                "Invalid argument given, PhpFileCache only allows objects that implement __set_state() " .
-                "and fully support var_export(). You can use the FilesystemCache to save arbitrary object " .
-                "graphs using serialize()/deserialize()."
-            );
-        }
-
         $filename  = $this->getFilename($id);
 
         $value = array(
             'lifetime'  => $lifeTime,
             'data'      => $data
         );
 
-        $value  = var_export($value, true);
-        $code   = sprintf('<?php return %s;', $value);
+        if (is_object($data) && method_exists($data, '__set_state')) {
+            $value  = var_export($value, true);
+            $code   = sprintf('<?php return %s;', $value);
+        } else {
+            $value  = var_export(serialize($value), true);
```

Comment:
Actually , this code calls `serialize` for an array, it only does simple `var_export` for object implementing `__set_state`. I did this way because I didn't want to traverse an array to check values' types.